app/core/WalletConnect/WalletConnectV2.test.ts

```diff
@@ -265,6 +269,222 @@ describe('WC2Manager', () => {
+      });
+
+      const pairedSessions = manager.getSessions();
+      expect(pairedSessions.length).toBeGreaterThan(0);
+
+      expect(storageSpy).toHaveBeenCalledWith(
+        AppConstants.WALLET_CONNECT.DEEPLINK_SESSIONS,
+        expect.any(String),
+      );
+    });
+
+    it('should handle existing sessionTopic connections', async () => {
+      const mockWcUri = 'wc://test@2?sessionTopic=test-topic';
+      const showLoadingSpy = jest.spyOn(wcUtils, 'showWCLoadingState');
+
+      await manager.connect({
+        wcUri: mockWcUri,
+        redirectUrl: 'https://example.com',
+        origin: AppConstants.DEEPLINKS.ORIGIN_DEEPLINK,
+      });
+
+      expect(showLoadingSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should handle WalletConnect v1 URIs', async () => {
+      const mockWcUri = 'wc:00e46b69-d0cc-4b3e-b6a2-cee442f97188@1';
+      const WalletConnectSpy = jest.spyOn(WalletConnect, 'newSession');
+
+      await manager.connect({
+        wcUri: mockWcUri,
+        redirectUrl: 'https://example.com',
+        origin: 'qrcode',
+      });
+
+      expect(WalletConnectSpy).toHaveBeenCalledWith(
+        mockWcUri,
+        'https://example.com',
+        false,
+        'qrcode'
+      );
+    });
+
+    it('should handle invalid URIs', async () => {
+      const mockWcUri = 'invalid:uri';
+      const consoleSpy = jest.spyOn(console, 'warn').mockImplementation();
+
+      await manager.connect({
+        wcUri: mockWcUri,
+        redirectUrl: 'https://example.com',
+        origin: 'qrcode',
+      });
+
+      expect(consoleSpy).toHaveBeenCalled();
+      consoleSpy.mockRestore();
+    });
+  });
+
+  describe('WC2Manager session request handling', () => {
+    let mockWeb3Wallet: IWalletKit;
+
+    beforeEach(() => {
+      mockWeb3Wallet = (manager as unknown as { web3Wallet: IWalletKit }).web3Wallet;
+    });
+
+    it('should handle session requests through event emission', async () => {
+      // Get the callback that was registered for 'session_request'
+      const sessionRequestCallback = (mockWeb3Wallet.on as jest.Mock).mock.calls.find(
+        ([event]) => event === 'session_request'
+      )?.[1];
+
+      expect(sessionRequestCallback).toBeDefined();
+
+      const mockRequest = {
+        topic: 'test-topic',
+        id: 1,
+        params: {
+          request: {
+            method: 'eth_sendTransaction',
+            params: [],
+          },
+        },
+        verifyContext: {
+          verified: {
+            verifyUrl: 'https://example.com',
+            validation: 'VALID' as const,
+            origin: 'https://example.com'
+          }
+        }
+      };
+
+      // Call the callback directly
+      await sessionRequestCallback(mockRequest);
+      
+      const session = manager.getSession('test-topic');
+      expect(session).toBeDefined();
+    });
+
+    it('should reject invalid session requests through event emission', async () => {
```

Comment:
S: what do you mean by "through event emission"?

```suggestion
    it('rejects invalid session requests through event emission', async () => {
```